.github/workflows/linux.yml

```diff
@@ -132,99 +119,157 @@ jobs:
+            -DOpenVINODeveloperPackage_DIR=${BUILD_DIR} \
+            -S ${OPENVINO_CONTRIB_REPO}/modules \
+            -B ${OPENVINO_CONTRIB_BUILD_DIR}
 
-      - name: Cmake Layer Tests
-        run: cmake -GNinja -S ${OPENVINO_REPO}/tests/layer_tests -B ${BUILD_DIR}/layer_tests
+          cmake --build ${OPENVINO_CONTRIB_BUILD_DIR} --parallel --config ${{ env.CMAKE_BUILD_TYPE }}
 
-      - name: Cmake Model Hub Tests
-        run: cmake -GNinja -S ${OPENVINO_REPO}/tests/model_hub_tests -B ${BUILD_DIR}/model_hub_tests
+      - name: Show ccache stats
+        run: ccache --show-stats
 
-      - name: Build Layer Tests
-        run: cmake --build ${BUILD_DIR}/layer_tests --parallel --config Release
+      - name: Cmake install - OpenVINO
+        run: |
+          cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/cmake_install.cmake
+          cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_TEST_DIR} -DCOMPONENT=tests -P ${BUILD_DIR}/cmake_install.cmake
+          cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DCOMPONENT=python_wheels -P ${BUILD_DIR}/cmake_install.cmake
 
-      - name: Build Model Hub Tests
-        run: cmake --build ${BUILD_DIR}/model_hub_tests --parallel --config Release
+      - name: Pack Artifacts
+        run: |
+          pushd ${INSTALL_DIR}
+            tar -czvf ${BUILD_DIR}/openvino_package.tar.gz *
+          popd
 
-      - name: Install wheel packages
-        run: cmake -DCOMPONENT=python_wheels -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/cmake_install.cmake
+          pushd ${INSTALL_TEST_DIR}
+            tar -czvf ${BUILD_DIR}/openvino_tests.tar.gz *
+          popd
 
-      - name: Install Layer Tests
-        run: cmake -DCOMPONENT=tests -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/layer_tests/cmake_install.cmake
+      #
+      # Upload build artifacts
+      #
 
-      - name: Install Model Hub Tests
-        run: cmake -DCOMPONENT=tests -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/model_hub_tests/cmake_install.cmake
+      - name: Upload openvino package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: openvino_package
+          path: ${{ env.BUILD_DIR }}/openvino_package.tar.gz
+          if-no-files-found: 'error'
 
-      - name: Install python wheels
-        run: python3 -m pip install openvino-dev --find-links=${INSTALL_DIR}/tools
+      - name: Upload openvino tests package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.BUILD_DIR }}/openvino_tests.tar.gz
+          if-no-files-found: 'error'
 
-      - name: Install tests
-        run: cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DCOMPONENT=tests -P ${BUILD_DIR}/cmake_install.cmake
+  Samples:
+    needs: Build
+    defaults:
+      run:
+        shell: bash
+    runs-on: ubuntu-20.04
+    container:
+      image: ubuntu:20.04
```

Comment:
if this job is using the `runs-on: ubuntu-20.04`, should it be in a container? Or is it for the paths' consistency?

Reply:
I want to run OpenVINO is a empty container to check all dependencies are installed by `install_openvino_depedencues.sh`